Source/Plugins/Core/com.equella.core/src/com/tle/web/controls/youtube/YoutubeUtils.java

```diff
@@ -55,6 +59,60 @@ public static String formatDuration(String duration) {
     return String.format(format, seconds, minutes, hours);
   }
 
+  /**
+   * Parses a date string returned from the Youtube Data API V3's "modified" entry for a given
+   * Youtube search result. Returns a date epoch long.
+   *
+   * @param date The string to parse (Expects it to be in ISO_DATE_TIME format)
+   * @return a long containing the date represented as the number of milliseconds since midnight
+   *     January 1, 1970.
+   * @throws RuntimeException when the date cannot be parsed.
+   */
+  public static Optional<Long> parseDateModifiedToMillis(String date) {
+    Optional<Long> parsedDate;
+    try {
+      parsedDate =
+          Optional.of(
+              ZonedDateTime.parse(date, DateTimeFormatter.ISO_DATE_TIME)
+                  .toInstant()
+                  .toEpochMilli());
+    } catch (DateTimeParseException e) {
+      throw new DateTimeParseException(
+          "Unable to parse youtube date modified", date, e.getErrorIndex());
+    }
+    return parsedDate;
+  }
+
+  /**
+   * Wrapper for parseDateModifiedToMillis which if present will return the date as a
+   * java.util.Date.
+   *
+   * <p>Youtube Data API v3 returns ISO_DATE_TIME strings, but existing attachments returned from v2
+   * will be stored as epoch millis, so this function supports both.
+   *
+   * @param date The date object to parse (Expects it to be a string in ISO_DATE_TIME format, or an
+   *     epoch long)
+   * @return The date represented as a java.util.Date.
+   * @throws IllegalArgumentException if passed in Object is not an instance of long or string
+   */
+  public static Optional<Date> parseDateModifiedToDate(Object date) {
+    Optional<Date> parsedDate = Optional.empty();
+    if (date instanceof Long) {
+      // if its a long, assume its an epoch long
+      parsedDate = Optional.of(new Date((Long) date));
+    } else if (date instanceof String) {
+      // assume its an ISO_DATE_TIME string
+      Optional<Long> parsedLong = parseDateModifiedToMillis((String) date);
+      if (parsedLong.isPresent()) {
+        parsedDate = Optional.of(new Date(parsedLong.get()));
+      }
```

Comment:
Is this not just something like:

```suggestion
      parsedDate = parseDateModifiedToMillis((String) date)
        map(millis -> new Date(millis));
```

I'm just freestyling there, but hopefully you get the idea.

Reply:
You were close! It has to be `flatMap`.